mypy_django_plugin/transformers/managers.py

```diff
@@ -115,6 +118,22 @@ def get_funcdef_type(definition: Union[FuncBase, Decorator, None]) -> Optional[P
     )
 
 
+def _find_type_var(queryset_info: TypeInfo, ret_type: TypeVarType) -> Optional[MypyType]:
+    """
+    Attempts to find the concrete type corresponding to a TypeVarType in a queryset's base types.
+
+    Example:
+    Suppose queryset_info is based on a generic class `QuerySet[T]` and ret_type corresponds
+    to `T`, then this function will return the concrete type that `T` was instantiated with.
+    """
+    for base in queryset_info.bases:
+        for i, type_var in enumerate(base.type.type_vars):
+            type_var_path = f"{base.type.module_name}.{type_var}"
+            if type_var_path == ret_type.fullname and i < len(base.args):
+                return base.args[i]
+    return None
```

Comment:
As you've touched in https://github.com/typeddjango/django-stubs/pull/1666#issuecomment-1700700518 I think we rather need to figure out the correct way to "forward" the `TypeVar`. As the argument could be used for another generic class, e.g. `list`, `Collection`, `Sequence`, `dict`, my custom generic class, etc.

We should cover such cases in in your test.

Reply:
I have added the code and the test in my last commit